src/main/java/pl/edu/mimuw/Main.java

```diff
@@ -3,60 +3,282 @@
 import javafx.application.Application;
 import javafx.scene.Scene;
 import javafx.scene.control.Button;
+import javafx.scene.control.Label;
 import javafx.scene.control.TextArea;
 import javafx.scene.layout.HBox;
 import javafx.scene.layout.VBox;
 import javafx.stage.Stage;
+import org.w3c.dom.Text;
+import pl.edu.mimuw.model.Expression;
+import pl.edu.mimuw.model.argumentless.Constant;
+import pl.edu.mimuw.model.argumentless.Placeholder;
+import pl.edu.mimuw.model.binary.Divide;
+import pl.edu.mimuw.model.binary.Minus;
+import pl.edu.mimuw.model.binary.Plus;
+import pl.edu.mimuw.model.binary.Times;
+import pl.edu.mimuw.model.unary.Cos;
+import pl.edu.mimuw.model.unary.Logarithm;
+import pl.edu.mimuw.model.unary.Sin;
 
-import java.util.function.Consumer;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Main extends Application {
-
   private static final double BUTTON_MIN_WIDTH = 72;
   private static final double BUTTON_MIN_HEIGHT = 72;
+  private static final int BUTTON_COLUMNS = 5;
+  private static final int BUTTON_ROWS = 5;
+  private static final String infoText = "Usage:\n" +
+    "Select an expression\n" +
+    "Build child expression\n" +
+    "When done, click LEFT DONE, CHILD DONE or RIGHT DONE\n" +
+    "to connect new expression to the one level higher.";
+
+  private static final List<Expression> expressionStack = new ArrayList<>();
```

Comment:
to czemu nie `private static final Stack<Expression> expressionStack = new Stack<>();`? 😉